homework/index.js

```diff
@@ -30,15 +34,147 @@
+        }
+        contributors.forEach(contributor => {
+          const li = createAndAppend('li', ul, { class: 'contributor-item' });
+          createAndAppend('img', li, {
+            src: contributor.avatar_url,
+            alt: contributor.login,
+            class: 'contributor-avatar',
+          });
+          const contributorData = createAndAppend('div', li, { class: 'contributor-data' });
+          createAndAppend('div', contributorData, { text: contributor.login });
+          createAndAppend('div', contributorData, {
+            text: contributor.contributions,
+            class: 'contribution-count',
+          });
+          li.addEventListener('click', () => {
+            window.open(contributor.html_url);
+          });
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', parents.forContributors, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      })
+      .then(() => {
+        loadingImg.style.display = 'none';
+      });
+  }
+
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('label', header, { for: 'select-menu', text: 'HYF Repositories' });
+    const selectMenu = createAndAppend('select', header, {
+      class: 'select-menu',
+      id: 'select-menu',
+    });
+    const container = createAndAppend('div', root, { class: 'container' });
+    const leftColumn = createAndAppend('div', container, { class: 'left column' });
+    const rightColumn = createAndAppend('div', container, { class: 'right column' });
+    const loadingImg = createAndAppend('img', root, {
+      src: './images/loading-image.gif',
+      class: 'loading',
+    });
+
+    fetchJSON(url)
+      .then(repos =>
+        repos
+          .sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()))
+          .map(element => ({
+            name: element.name,
+            description: element.description,
+            forks: element.forks,
+            updated: element.updated_at,
+            repoUrl: element.html_url,
+            contributorsUrl: element.contributors_url,
+          })),
+      )
+      .then(repos => {
+        repos.forEach((repo, index) => {
+          createAndAppend('option', selectMenu, {
+            text: repo.name,
+            value: index,
+          });
+        });
+
+        showRepoAndContributorsInDetail(repos[selectMenu.value], {
+          forRepo: leftColumn,
+          forContributors: rightColumn,
+        });
+
+        selectMenu.addEventListener('change', () => {
+          showRepoAndContributorsInDetail(repos[selectMenu.value], {
+            forRepo: leftColumn,
+            forContributors: rightColumn,
+          });
+        });
+      })
+      .catch(error => {
+        createAndAppend('div', root, { text: error.message, class: 'alert-error' });
+      })
+      .then(() => {
```

Comment:
Nice touch to add a spinner. This is exactly the right place to add the required `then` as it will be executed in the normal case as well as the error case. You also may want to take note of the `.finally()` method that can be used at the end of a promise chain, although at this time it is not yet universally supported by the various browsers.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally